packages/react-ui-core/src/Modal/ClosableModal.js

```diff
@@ -0,0 +1,72 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import autobind from 'autobind-decorator'
+
+import { Modal } from '../Modal'
+
+@themed(/^ClosableModal/)
+export default class ClosableModal extends Component {
```

Comment:
I'm not getting the purpose of this component. Why can't this already be done in the existing Modal?

Reply:
Modal does not have state for `isOpen` and we are setting ourselves up to implement state of isOpen in many of our uses of `Modal`.

Reply:
I'd rather have this functionality in `Modal` TBH.  This extra component is just going to confuse people.

Reply:
The only check you'd have to add would be from [this line](https://github.com/rentpath/react-ui/pull/217/files#diff-53fc33bc9837831b8b065721991107dbR42): 

```
if (this.props.onClose) {
  this.setState({  isOpen: false })
  this.props.onClose()
}
```